s3/src/main/resources/reference.conf

```diff
@@ -125,4 +125,20 @@ pekko.connectors.s3 {
 
   # Add signature headers to requests when aws.credentials.provider is anon
   sign-anonymous-requests = true
+
+  additional-allowed-headers {
+    GetObject = []
```

Comment:
I might be missing something, but are these the right defaults? It feels like by default there should be some entries here (since you are testing them) and if someone doesn't want these additional allowed headers then they can override it in their own `.conf` with an empty list.